### PR TITLE
File formats: Uppercase file types when importing a file.

### LINKED ISSUE #10879: Inconsistent import/export file extension handling
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When using File>Import or File>Export the dialogs handle file extensions differently and don't handle typical file extensions.

In addition, import/export is not consistent for different format selections.

### Full version info

```shell
Many available downloads use, for example .STEP. The import dialog for step import sees files with the extension .step, but not .STEP (this is the same for .dxf/.DXF, .iges/.IGES, etc.)

The export dialog, when no expression is specified with dxf selected, will export a file with no extension. (The import dialog then ignores the file.

If, for example, the last time export was used step format was selected and extension provided, then dxf is selected for the type, the extension is not updated to dxf.

See images below for screen shots of each case described.

This was seen in previous versions and the images shown are from this version:
[code]
OS: Debian GNU/Linux 12 (bookworm) (XFCE/xfce)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.34479 (Git)
Build type: Release
Branch: master
Hash: c87860922c90007f71de9f1886ae8176e00ac977
Python 3.11.2, Qt 5.15.8, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * MnesarcoUtils 0.2.2
  * ZolkoBase 0.0.1
  * lattice2 1.0.0
  * Curves 0.6.13
  * Nodes 0.1.36
  * Help 1.0.3
  * QuickMeasure 2022.10.28
  * Manipulator 1.5.0
  * Assembly
  * Assembly4 0.50.2
  * freecad.gears 1.0.0
  * sheetmetal 0.3.0
  * Silk 0.1.5
  * kicadStepUpMod 10.19.4
  * A2plus 0.4.60n
  * DynamicData 2.47.0
  * workfeature-macro
  * fasteners 0.4.66
  * ExtremeProDark 2.9.0
  * Defeaturing 1.2.0
  * CurvedShapes 1.0.4
[/code]
```

### Subproject(s) affected?

None

### Anything else?

DxfSelected
DxfExportNoExtensionSpecified
DxfImport

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Predko**
Fixes #10879

Adds uppercase file types to the file import filter.

### CODE REVIEW

**Review by yorikvanhavre — APPROVED** [review]

File: src/Gui/FreeCADGuiInit.py [hunk 1]
```diff
@@ -301,9 +301,9 @@ def _SoGroup_init(self,*args):
 Gui.activateWorkbench("NoneWorkbench")
 
 # Register .py, .FCScript and .FCMacro
-FreeCAD.addImportType("Inventor V2.1 (*.iv)","FreeCADGui")
-FreeCAD.addImportType("VRML V2.0 (*.wrl *.vrml *.wrz *.wrl.gz)","FreeCADGui")
-FreeCAD.addImportType("Python (*.py *.FCMacro *.FCScript)","FreeCADGui")
+FreeCAD.addImportType("Inventor V2.1 (*.iv *.IV)","FreeCADGui")
+FreeCAD.addImportType("VRML V2.0 (*.wrl *.WRL *.vrml *.VRML *.wrz *.WRZ *.wrl.gz *.WRL.GZ)","FreeCADGui")
+FreeCAD.addImportType("Python (*.py *.PY *.FCMacro *.FCScript)","FreeCADGui")
```
Comment (wwmayer) [inline src/Gui/FreeCADGuiInit.py:306]: PY isn't a valid file extension for Python modules.
`touch /tmp/mytest.PY`
Start FreeCAD or Python interpreter:
```
import sys
sys.path.append("/tmp")
import mytest
```
==> 
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/shiboken2/files.dir/shibokensupport/__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'mytest'
```

Comment (Predko) [reply src/Gui/FreeCADGuiInit.py]: Sorry, I didn't know)

Comment (adrianinsaval) [reply src/Gui/FreeCADGuiInit.py]: maybe also add *.fcmacro and *.fcscript?

Comment (Predko) [reply src/Gui/FreeCADGuiInit.py]: @adrianinsaval ,
Files with these extensions are written only by FreeCAD and only in this form. There's no point in changing it.

Comment (adrianinsaval) [reply src/Gui/FreeCADGuiInit.py]: manually saved scripts from a generic text editor may have just lowercase extensions

Comment (Predko) [reply src/Gui/FreeCADGuiInit.py]: @adrianinsaval,
Added *.fcmacro *.fcscript.

File: src/Gui/Application.cpp [hunk 2]
```diff
@@ -344,7 +344,7 @@ namespace {
         std::stringstream str;
         str << "Image formats (";
         for (const auto& ext : supportedFormats) {
-            str << "*." << ext.constData() << " ";
+        str << "*." << ext.constData() << " *." << ext.toUpper().constData() << " ";
```
Comment (adrianinsaval) [inline src/Gui/Application.cpp:347]: the indentation should have been kept I think

Comment (adrianinsaval) [inline src/Gui/Application.cpp:347]:
```suggestion
            str << "*." << ext.constData() << " *." << ext.toUpper().constData() << " ";
```

Comment (Predko) [reply src/Gui/Application.cpp]: Yes, I didn't notice.

**Pull request merged**